kpatch-build/create-diff-object.c

```diff
@@ -2695,13 +2704,22 @@ void kpatch_create_dynamic_rela_sections(struct kpatch_elf *kelf,
 				 * patched.
 				 */
 				if (lookup_global_symbol(table, rela->sym->name,
-							 &result))
+							 &result)) {
 					/*
-					 * Not there, assume it's either an
-					 * exported symbol or provided by
-					 * another .o in the patch module.
+					 * Not there, see if the symbol is
+					 * exported, and set sym_objname to the
+					 * object the exported symbol belongs to
 					 */
-					external = 1;
+					tmp_objname = find_exported_symbol_objname(mod_symvers,
+										   rela->sym->name);
+					if (tmp_objname) {
+						 /* Strip the path, only want the module name */
+						sym_objname = basename(tmp_objname);
```

Comment:
The file name needs to be converted to the "module name" by converting dashes ('-') to underlines ('_').